cloud/amazon/ec2_vpc_net.py

```diff
@@ -144,21 +144,28 @@ def update_vpc_tags(vpc, module, vpc_obj, tags, name):
 
     tags.update({'Name': name})
     try:
-        current_tags = dict((t.name, t.value) for t in vpc.get_all_tags(filters={'resource-id': vpc_obj.id}))
-        if cmp(tags, current_tags):
-            vpc.create_tags(vpc_obj.id, tags)
-            return True
-        else:
-            return False
+        if vpc_obj:
+            current_tags = dict((t.name, t.value) for t in vpc.get_all_tags(filters={'resource-id': vpc_obj.id}))
+            if cmp(tags, current_tags):
+                if not module.check_mode:
+                    vpc.create_tags(vpc_obj.id, tags)
+                else:
+                    vpc_obj.tags = tags
```

Comment:
Maybe I'm missing something but shouldn't this be updated regardless whether check mode is enabled?  The final return values seem to pull from vpc_obj only.